### DISCUSSION TITLE
Access Modal State

### DISCUSSION THREAD

**Discussion by pkl-kirito** [saadeghi/daisyui#1864] [General]
Hi, how to access the state of the modal is open or close ? Because I have a separate modal, and useEffect only trigger once on page load, but I want to trigger the `useEffect` every time modal is open.

```
function HistoryModal(){

    useEffect(() => {
        console.log(`effect`);
        return () => {
        }
    }, []);


    return (
        <>
            <input type="checkbox" id="history-modal" className="modal-toggle" />
            <label htmlFor="history-modal" className="modal">
                <label  className="w-11/12 max-w-5xl backdrop-blur-sm bg-modalBlack bg-opacity-80 rounded-2xl border border-stroke">
                    
                </label>
            </label>
        </>
    )
}

export default HistoryModal
```

**Comment by saadeghi** [comment]
Here the value of checkbox defines if the modal is open or not so in React you can bind a value to checkbox and then open/close the checkbox as you need.

```jsx
const [isChecked, setIsChecked] = useState(false)
```
```jsx
<input type="checkbox" id="history-modal" className="modal-toggle" checked={isChecked} />
```

Let me know if you have any questions

**Reply by pkl-kirito** [reply]
i saw v3.0.0 is using dialog, how to get the dialog state is open or close ?

**Reply by fini** [reply]
[HTMLDialogElement: open property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLDialogElement/open)
